spec/unit/coinbase/smart_contract_spec.rb

```diff
@@ -198,6 +198,1229 @@
+                                                })
+          )
+        end
+
+        it 'returns the parsed int64 value' do
+          expect(result).to eq(-9_223_372_036_854_775_808)
+        end
+      end
+
+      describe 'int128' do
+        let(:method_name_for_context) { 'getInt128' }
+
+        before do
+          allow(smart_contracts_api).to receive(:read_contract).and_return(
+            Coinbase::Client::SolidityValue.new({
+                                                  'type' => 'int128',
+                                                  'value' => '-170141183460469231731687303715884105728'
+                                                })
+          )
+        end
+
+        it 'returns the parsed int128 value' do
+          expect(result).to eq(-170_141_183_460_469_231_731_687_303_715_884_105_728)
+        end
+      end
+
+      describe 'int256' do
+        let(:method_name_for_context) { 'getInt256' }
+        let(:int256_min) do
+          '-57896044618658097711785492504343953926634992332820282019728792003956564819968'
+        end
+
+        before do
+          allow(smart_contracts_api).to receive(:read_contract).and_return(
+            Coinbase::Client::SolidityValue.new({
+                                                  'type' => 'int256',
+                                                  'value' => int256_min
+                                                })
+          )
+        end
+
+        it 'returns the parsed int256 value' do
+          min_value =
+            -57_896_044_618_658_097_711_785_492_504_343_953_926_634_992_332_820_282_019_728_792_003_956_564_819_968
+          expect(result).to eq(min_value)
+        end
+      end
+    end
+
+    describe 'boolean type' do
+      let(:method_name_for_context) { 'pureBool' }
+
+      before do
+        allow(smart_contracts_api).to receive(:read_contract).and_return(
+          Coinbase::Client::SolidityValue.new({
+                                                'type' => 'bool',
+                                                'value' => 'true'
+                                              })
+        )
+      end
+
+      it 'returns the parsed boolean value' do
+        expect(result).to be(true)
+      end
+    end
+
+    describe 'address type' do
+      let(:method_name_for_context) { 'pureAddress' }
+      let(:address) { '0xd8da6bf26964af9d7eed9e03e53415d37aa96045' }
+
+      before do
+        allow(smart_contracts_api).to receive(:read_contract).and_return(
+          Coinbase::Client::SolidityValue.new({
+                                                'type' => 'address',
+                                                'value' => address
+                                              })
+        )
+      end
+
+      it 'returns the parsed address value' do
+        expect(result).to eq(address)
+      end
+    end
+
+    describe 'array type' do
+      let(:method_name_for_context) { 'pureArray' }
```

Comment:
Can do table-driven test scenarios like the following to reduce on some of the setup overhead:
```
[
  {test: 'uint256', method_name: 'getUint256', solidity_value: {type: 'uint256', value: '1'}, expected_value: 1},
  {
      test: 'array type'
      method_name: 'pureArray',
      solidity_value: Coinbase::Client::SolidityValue.new(
      type: 'array',
      values: [
         Coinbase::Client::SolidityValue.new(type: 'uint256', value: '1'),
         Coinbase::Client::SolidityValue.new(type: 'uint256', value: '2'),
         Coinbase::Client::SolidityValue.new(type: 'uint256', value: '3'),
      ]
    },
    expected_value: [1, 2, 3]
  }
].each do |test|
  context "when the return value is: `#{test[:solidity_value].test}`" do
    let(:method_name) { test[:method_name] }

    before do
      allow(smart_contracts_api).to receive(:read_contract).and_return(test[:solidity_value])
    end

    it 'returns the expected value' do
      expect(result).to eq(test[:expected_value])
    end
  end
end
```

Reply:
yeah this is clean, refactoring to it